docs/wiki/Getting started - Scenario 2 Onboard module library and CI environment.md

```diff
@@ -136,6 +144,7 @@ To use the environment's pipelines you should use the information you gathered d
 | `ARM_TENANT_ID` | `9734cec9-4384-445b-bbb6-767e7be6e5ec` | The tenant ID of the Azure Active Directory tenant to test-deploy modules in. |
 | `AZURE_CREDENTIALS` | `{"clientId": "4ce8ce4c-cac0-48eb-b815-65e5763e2929", "clientSecret": "<placeholder>", "subscriptionId": "d0312b25-9160-4550-914f-8738d9b5caf5", "tenantId": "9734cec9-4384-445b-bbb6-767e7be6e5ec" }` | The login credentials of the deployment principal used to log into the target Azure environment to test in. The format is described [here](https://github.com/Azure/login#configure-deployment-credentials). |
 | `PLATFORM_REPO_UPDATE_PAT` | `<placeholder>` | A private access token (PAT) with enough permissions assigned to it to push into the main branch. This PAT is leveraged by pipelines that automatically generate ReadMe files to keep them up to date. |
+| `TOKEN_NAMEPREFIX` | `<cntso>` | Optional. If you specify the name prefix token here, this will take precedence over the `localToken_namePrefix` specified in the `settings.yml`.  |
```

Comment:
Didn't we describe it the opposite way in the issue?  -  If specified in the file it has priority? Also seems more obvious to me as it is more transparent & explicit. Or I'm mixing things up,

Reply:
Yes that was the initial suggestion from @eriqua, and yes I have flipped it around based on the demo sessions we had > 2 months ago, but I don't remember there were other opinions as I had it articulated here https://github.com/Azure/ResourceModules/issues/1440#issuecomment-1151033577. 

However, I think it depends on what the path we want to provide to customers as a starting point:

1) Prioritize the **remote** (secret/variable) `namePrefix` token. And also set the local one to a dummy value.
   -  Here we do not change the existing mechanism and support both models. So even we rename our current token locally to be called `cntso`. Then that is not impacting our CI. Also guide the users to change this value to their preferred token name when testing locally.
   - Even if users fork carml and not change the namePrefix locally, they don't impact our CI. 

  2) Prioritize the **local** `namePrefix` token. And also set it to `''` as default on file. Making it explicit change for users:
     - Here we are defaulting the user experience to the remote (secret/variable) `namePrefix` token via the getting started guide.
     - Here we have to remove our local `namePrefix` token and set the value to `''`. Also guide the users to go to the `settings.yml` and change the `''` to `'TheirNamePrefixTokenValue'` to allow for local testing.

3) Completely remove the **local** `namePrefix` token and default the user experience to the **remote** (secret/variable) `namePrefix` token

   - Remove support for the local file option and replace it with placeholder tokens for the sake of examples. 
   - Here we are introducing a major change, and we need to guide the users for local testing mechanism. Example: inject an additional token called 'namePrefix' with the Test-ModuleLocally script. 

After giving it additional thought as per the above and with the current implementation of option (1) with this PR. Would this require us to have another session discussing this? FYI, changing to option (2) [original description], is not hard as well.

Reply:
I saw we also have an open comment from @eriqua in this PR somwhere towards the top. She's definitely vouching for Option 1, while I'm of the opinion that Option 2 is the more intuitive way.

If you want we can also have another chat later today. 

I somewhat have this scenario in my head that somebody has an error, tries to troubleshoot it, sees that there is the name prefix specified in the `settings.yml` and does it no see reflected in Azure. Why? Well because there is also a secret this person cannot see that renders the local configuration meaningless. Good luck for that poor contributor to figure this out :D 

Also I can't stress enough that this feature is something we implement primarily for us to avoid conflicts with other forks. I'd expect that new joiners will go ahead and use the `settings.yml` file rather than the secret to setup the prefix as it's not only more transparent, but also simply more obvious. I wouldn't want to push people to use the secrets mechanism for something that isn't a secret for no good reason. 

So, to summarize - my opinion is that we should
- Support both the secret & `settings.yml`
- Use the secret on our end and leave the value in the `settings.yml` file empty
- Let the `settings.yml` file take priority if both are specified

If you have some time today, I'm also up for a chat. I'll also try and ping some of the other contributors to get their thoughts.

Reply:
I actually vouch for option 2, as described in the related issue. In particular, **I definitely vote for keeping the local token empty**.
- Keeping `carml` as an example, even if we adopt the remote secret, will keep generating issues with having the global unique names taken, which is the primary reason for which we're implementing this change, and I'm afraid all this great work would be useless.
- Keeping an example token, such as `cntso`, will solve the issue on our end, but will cause the same conflict among users adopting the CI environment, because let's consider that unfortunately not everyone follows the documentation in detail and many will stay with what is already set. It is what it is.

So my suggestion is to move to option 2, support both local and remote, keep the local token empty, agree that having the local one taking precedence if both are set makes sense, for the reasons described by @MrMCake  above.

The only point on which I don't 100% agree is that we're implementing this change only for us. Unfortunately the remote token in GitHub is only possible as a secret and not as a variable, as ADO allows. And I agree this makes setting a secret for something which is not a secret confusing. On the other hand, I believe the implementation is helping the contribution flow, not only our maintenance. Using the remote secret, contributors will be able to leave the local token empty, so avoiding to make a last change before opening PRs from their fork, something reviewers need to double check, which may be error prone especially when PRs are touching a lot of files. That's why in my other comment I suggest to be more vocal in leading users (especially contributors) to use the remote token as an option. And that's also why I'd like to find a way to avoid changing the container registry unique name in the settings.yml as well, but not in this PR.

Reply:
Sounds good @eriqua and thanks for jumping in as I know you're away. I know @MrMCake is also trying to get a hold of another opinion. Nonetheless, I am also in favour for option 2 now as it's much better clarity on the rationale. Will adjust and test to ensure we don't miss on something and also update the documentation explicitly call this out as requirement for customers for the CI to function (if they don't opt in for a remote/secret token), and perform local testing as well.

Will give it some thought on the continent registry after this PR.

Reply:
@eriqua, thanks for jumping in. Then I misunderstood the other statement. Agree with everything you stated above.

Reply:
@eriqua @MrMCake updated to perform the following:
- Out of the box cloning experience for customers will have the `localToken_namePrefix` set to `''`. 
- Customers will use the getting started guide to add their name prefix token in the settings.yml.
- Customers may also choose to leave it empty as the default, but the pipeline will fail if they do not provide the remote version of the namePrefix token.
- To perform local testing. Customers need to either: 
  - add an additional token to the Test-ModuleLocally script to include a name prefix.
  - OR update the `localToken_namePrefix` to include the name prefix they want to use.
- If customers leave the `localToken_namePrefix` in the settings.yml with their own value. The pipeline will prioritize it even if the remote/secret version is enabled.

So TL;DR I implemented option (2) 

@MrMCake can you please create the 'TOKEN_NAMEPREFIX' with our value in the ADO variable group so our ADO pipelines do not fail

Reply:
@ahmadabdalla done, thanks